src/test/java/com/exasol/adapter/dialects/impala/ImpalaSqlDialectTest.java

```diff
@@ -109,16 +112,28 @@ void testValidateCatalogProperty() throws PropertyValidationException {
 
     @Test
     void testValidateSchemaProperty() throws PropertyValidationException {
-        setMandatoryProperties("IMPALA");
+        setMandatoryProperties();
         this.rawProperties.put(SCHEMA_NAME_PROPERTY, "MY_SCHEMA");
         final AdapterProperties adapterProperties = new AdapterProperties(this.rawProperties);
         final SqlDialect sqlDialect = new ImpalaSqlDialect(null, adapterProperties);
         sqlDialect.validateProperties();
     }
 
-    @Test
-    void testApplyQuote() {
-        assertThat(this.dialect.applyQuote("tableName"), equalTo("`tableName`"));
+    @CsvSource({ "tableName, `tableName`", //
+            "`tableName, ```tableName`" //
+    })
+    @ParameterizedTest
+    void testApplyQuote(final String unquoted, final String quoted) {
+        assertThat(this.dialect.applyQuote(unquoted), equalTo(quoted));
+    }
+
+    @ValueSource(strings = { "ab:'ab'", "a'b:'a\\'b'", "a''b:'a\\'\\'b'", "'ab':'\\'ab\\''" })
+    @ParameterizedTest
+    void testGetLiteralString(final String definition) {
+        final int colonPosition = definition.indexOf(':');
```

Comment:
Please extract this code. Too much duplication.

Reply:
We are going to split dialects soon. We need this test for each dialect and I think it will make things harder if I extract this now